### PR TITLE
feat: add position props for segmented component

### LINKED ISSUE #50519: Segmented组件增加 竖起排列  主要是好看
### What problem does this feature solve?

好看

### What does the proposed API look like?

position = “xxxx”

### PR DESCRIPTION
**Pull request by liangchaofei**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
close: https://github.com/ant-design/ant-design/issues/50519

rc-segmented : https://github.com/react-component/segmented/pull/257
> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### CODE REVIEW

File: components/segmented/demo/position.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+垂直方向。
+
+## en-US
+
+vertical direction
```
Comment (afc163) [inline components/segmented/demo/position.md:7]:
```suggestion
Make it vertical.
```

File: components/segmented/demo/position.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Segmented } from 'antd';
+
+const Demo: React.FC = () => (
+  <Segmented<string>
+    position="vertical"
```
Comment (afc163) [inline components/segmented/demo/position.tsx:6]: type="vertical" 还是 direction="vertical" 还是 mode="vertical"?

Comment (liangchaofei) [reply components/segmented/demo/position.tsx]: > type="vertical" 还是 direction="vertical" 还是 mode="vertical"?

改成mode了，rc-segmented有个属性是direction

Comment (liangchaofei) [reply components/segmented/demo/position.tsx]: 改成vertical?: boolean

**Comment by afc163** [comment]
ci 挂了

File: components/segmented/index.en-US.md [hunk 3]
```diff
@@ -46,17 +47,18 @@ Common props ref：[Common props](/docs/react/common-props)
...
-| icon  | Display icon for Segmented item  | ReactNode | - |  |
-| disabled | Disabled state of segmented item | boolean | false |  |
-| className | The additional css class | string | - |  |
+| Property  | Description                      | Type             | Default | Version |
+| --------- | -------------------------------- | ---------------- | ------- | ------- |
+| label     | Display text for Segmented item  | ReactNode        | -       |         |
+| value     | Value for Segmented item         | string \| number | -       |         |
+| icon      | Display icon for Segmented item  | ReactNode        | -       |         |
+| disabled  | Disabled state of segmented item | boolean          | false   |         |
+| className | The additional css class         | string           | -       |         |
```
Comment (afc163) [inline components/segmented/index.en-US.md:61]: 不修改这个格式。

File: components/segmented/index.zh-CN.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ demo:
 
 <!-- prettier-ignore -->
 <code src="./demo/basic.tsx">基本</code>
+<code src="./demo/mode.tsx">垂直方向</code>
```
Comment (afc163) [inline components/segmented/index.zh-CN.md:24]:
```suggestion
<code src="./demo/vertical.tsx">垂直方向</code>
```

Comment (liangchaofei) [reply components/segmented/index.zh-CN.md]: done

File: components/segmented/index.en-US.md (hunk 3 above)
Comment (liangchaofei) [reply components/segmented/index.en-US.md]: done

File: components/segmented/index.zh-CN.md [hunk 5]
```diff
@@ -47,8 +48,9 @@ demo:
 | defaultValue | 默认选中的值 | string \| number |  |  |
 | disabled | 是否禁用 | boolean | false |  |
 | onChange | 选项变化时的回调函数 | function(value: string \| number) |  |  |
-| options | 数据化配置选项内容  | string\[] \| number\[] \| SegmentedItemType\[] | [] |  |
+| options | 数据化配置选项内容 | string\[] \| number\[] \| SegmentedItemType\[] | [] |  |
 | size | 控件尺寸 | `large` \| `middle` \| `small` | `middle` |  |
+| vertical | 排列方向 | `false` \| `true` | `false` |  |
```
Comment (afc163) [inline components/segmented/index.zh-CN.md:53]:
```suggestion
| vertical | 排列方向 | boolean | `false` |  |
```

File: components/segmented/index.tsx [hunk 6]
```diff
@@ -114,6 +116,7 @@ const InternalSegmented = React.forwardRef<HTMLDivElement, SegmentedProps>((prop
       ref={ref}
       prefixCls={prefixCls}
       direction={direction}
+      vertical={vertical}
```
Comment (afc163) [inline components/segmented/index.tsx:119]: 这个不需要，restProps 里会透传过去。

**Comment by afc163** [comment]
有冲突

File: package.json [hunk 7]
```diff
@@ -329,7 +329,7 @@
     "tar": "^7.4.0",
     "tar-fs": "^3.0.6",
     "terser": "^5.31.2",
-    "tsx": "4.19.0",
+    "tsx": "4.11.2",
```
Comment (afc163) [inline package.json:332]: 不应该有这些 diff

**Comment by afc163** [comment]
应该发到 feature 分支。

**Comment by liangchaofei** [comment]
> 应该发到 feature 分支。

新开了个pr: https://github.com/ant-design/ant-design/pull/50708 这个关闭了

**Pull request closed without merging**